src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1595,81 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        public static ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, IDebugModule2 module, bool isLoad)
```

Comment:
>bool isLoad [](start = 116, length = 11)

You actually need three states here - load, list, and unload. Probably better to pass in the module id as an argument.

Reply:
Is there a reason we need list as well?

Reply:
'list' (or better yet pulling out the code that figures out the module id from this function) is required for 'HandleModulesRequestAsync' -- right now your code is treating that as an add, which is wrong.

---
In reply to: [505074678](https://github.com/microsoft/MIEngine/pull/1054#discussion_r505074678) [](ancestors = 505074678)

Reply:
VSCodeDebugAdapter hosts sends a `ModulesRequest` on the start of an `attach`. 

This is in `RequestInitialModules`, so these modules should be `new`.